plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListenerModule.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage;
+
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.openlineage.client.OpenLineageClient;
+import io.trino.plugin.openlineage.config.OpenLineageClientHttpTransportConfig;
+import io.trino.plugin.openlineage.config.OpenLineageListenerConfig;
+import io.trino.plugin.openlineage.transport.OpenLineageConsoleTransport;
+import io.trino.plugin.openlineage.transport.OpenLineageHttpTransport;
+import io.trino.plugin.openlineage.transport.OpenLineageTransport;
+
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class OpenLineageListenerModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(OpenLineageListener.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(OpenLineageListenerConfig.class);
+
+        install(conditionalModule(
+                OpenLineageListenerConfig.class,
+                config -> config.getTransport().equals(io.trino.plugin.openlineage.OpenLineageTransport.CONSOLE),
+                conditionalBinder -> {
```

Comment:
nit: It could be internalBinder and we use

                conditionalBinder -> conditionalBinder.bind(OpenLineageTransport.class).to(OpenLineageConsoleTransport.class)));